File: src/backend/api/app/schemas/event.py
# app/schemas/event.py
from pydantic import BaseModel

class Score(BaseModel):
    visibility: float
    visible_time_ratio: float
    sky_glow: float
    moon_fract_illumi: float
    rain: float
    cloud: float
    met_visibility: float

class Event(BaseModel):
    location_name: str
    start_time: str
    end_time: str
    scores: Score
    event_type: str
    lat: float
    lon: float
    international_designators: list[str]

    class Config:
        from_attributes = True

class EventResponse(BaseModel):
    total: int
    events: list[Event]

File: src/frontend/src/types/event.ts
// src/types/event.ts
export interface Score {
  visibility: number;
  visible_time_ratio: number;
  sky_glow: number;
  moon_fract_illumi: number;
  rain: number;
  cloud: number;
  met_visibility: number;
}

export interface Event {
  location_name: string;
  start_time: string;
  end_time: string;
  scores: Score;
  event_type: string;
  lat: number;
  lon: number;
  international_designators: string[];
  horizon_profile: number[]; // 天球シミュレータで使用
}

export interface EventResponse {
  total: number;
  events: Event[];
}
